pkg/sync/sync.go

```diff
@@ -342,10 +342,15 @@ func doCopySingle(src, dst object.ObjectStorage, key string, size int64) error {
 		in, err = src.Get(key, 0, -1)
 		if err != nil {
 			if _, e := src.Head(key); e != nil {
-				logger.Debugf("Head src %s: %s", key, err)
-				copied.IncrInt64(-1)
-				copiedBytes.IncrInt64(size * (-1))
-				return nil
+				for _, eStr := range []string{"404", "not found", "not exist", "no such file or directory"} {
```

Comment:
it's better to double check it.

Reply:
I think we should return a special error from object store, rather than check the message here, because different object store may return same message but have different meanings.

Reply:
check error in all head functions ？

Reply:
We can do that for most common ones first